src/BinaryInstaller.php

```diff
@@ -206,7 +206,7 @@ protected function installBinary($binary, $version, $url, $sha, $hashalgo = 'sha
         else {
             $cacheFolder = sys_get_temp_dir() . \DIRECTORY_SEPARATOR . $binary . \DIRECTORY_SEPARATOR . $version;
         }
-        $cacheDestination = $cacheFolder . \DIRECTORY_SEPARATOR . $fileName;
+        $cacheDestination = $cacheFolder. \DIRECTORY_SEPARATOR . bin2hex(random_bytes(5)) . $fileName;
```

Comment:
Won't this mean we effectively invalidate the cache every time, and never use it?

Reply:
Yes. I had to do this because there was an error building Tugboat.
This error happened while running the composer install.
I couldn't find any other way.

Reply:
Would it make sense to only do the random string cache file after we recognize the error that is causing us to do this. This way we still get the benefits of cache for instances where everything is working fine?